### PR TITLE
Enum suggestion desc

### PR DESCRIPTION
**Pull request by karimra**
#157

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -50,8 +50,20 @@ const (
 	msgSize = 512 * 1024 * 1024
 )
 
-var encodings = []string{"json", "bytes", "proto", "ascii", "json_ietf"}
-var formats = []string{"json", "protojson", "prototext", "event", "proto"}
+var encodings = [][2]string{
+	{"json", "indicates that the value is encoded as a JSON string,as specified in RFC7159"},
```
Comment (hellt) [inline cmd/root.go:54]: I'd keep it more concise, like

`JSON encoded string (RFC7159)`

File: cmd/root.go [hunk 2]
```diff
@@ -50,8 +50,20 @@ const (
...
+	{"bytes", "contains a byte sequence whose semantics is opaque to the protocol"},
```
Comment (hellt) [inline cmd/root.go:55]: byte sequence whose semantics is opaque to the protocol

File: cmd/root.go [hunk 3]
```diff
@@ -50,8 +50,20 @@ const (
...
+	{"proto", "contains a serialised protobuf message using protobuf.Any"},
```
Comment (hellt) [inline cmd/root.go:56]: serialised protobuf message using protobuf.Any

File: cmd/root.go [hunk 4]
```diff
@@ -50,8 +50,20 @@ const (
...
+	{"ascii", "an ASCII encoded string representing text formatted according to a target-defined convention"},
```
Comment (hellt) [inline cmd/root.go:57]: I'd remove "an" in the beginning

File: cmd/root.go [hunk 5]
```diff
@@ -50,8 +50,20 @@ const (
...
+	{"json_ietf", "similar to `json` with aditional types defined"},
```
Comment (hellt) [inline cmd/root.go:58]: JSON_IETF encoded string (RFC7951)

File: cmd/root.go [hunk 6]
```diff
@@ -50,8 +50,20 @@ const (
...
+}
+var formats = [][2]string{
+	{"json", "similar to protojson but with xpath style paths and decoded timestamps"},
+	{"protojson", "protocol buffer messages as JSON format"},
+	{"prototext", "protocol buffer messages as the textproto format"},
+	{"event", "protocol buffer messages as a timestamped list of tags and values"},
+	{"proto", "protocol buffer messages as the binary wire format"},
```
Comment (hellt) [inline cmd/root.go:65]: `as -> in`?

protocol buffer messages as JSON format -> protocol buffer messages in JSON format
reads more naturally to me

File: cmd/subscribe.go [hunk 7]
```diff
@@ -36,8 +36,16 @@ import (
 
 const defaultRetryTimer = 10 * time.Second
 
-var subscriptionModes = []string{"once", "stream", "poll"}
-var streamSubscriptionModes = []string{"target-defined", "sample", "on-change"}
+var subscriptionModes = [][2]string{
+	{"once", "a single request/response channel. The target creates the relevant update messages, transmits them, and subsequently closes the RPC"},
+	{"stream", "long-lived subscriptions which continue to transmit updates relating to the set of paths that are covered within the subscription indefinitely"},
+	{"poll", "used for on-demand retrieval of data items via long-lived RPCs"},
```
Comment (hellt) [inline cmd/subscribe.go:42]: I'd remove "used for"

File: cmd/root.go (hunk 5 above)
Comment (karimra) [reply cmd/root.go:58]: is it not this RFC ? https://tools.ietf.org/html/rfc7493

File: cmd/root.go (hunk 6 above)
Comment (karimra) [reply cmd/root.go:65]: copied the package description from here: https://godoc.org/google.golang.org/protobuf/encoding/protojson

**Pull request merged**